src/tribler-core/tribler_core/components/tag/rules/tag_rules_processor.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from typing import Optional, Set
+
+from ipv8.taskmanager import TaskManager
+
+from pony.orm import db_session
+
+import tribler_core.components.metadata_store.db.orm_bindings.torrent_metadata as torrent_metadata
+import tribler_core.components.metadata_store.db.store as MDS
+from tribler_core.components.metadata_store.db.serialization import REGULAR_TORRENT
+from tribler_core.components.tag.community.tag_payload import TagOperation, TagOperationEnum
+from tribler_core.components.tag.db.tag_db import (
+    CLOCK_START_VALUE, PUBLIC_KEY_FOR_AUTO_GENERATED_TAGS,
+    TagDatabase,
+)
+from tribler_core.components.tag.rules.tag_rules import extract_only_valid_tags
+from tribler_core.notifier import Notifier
+
+LAST_PROCESSED_TORRENT_ID = 'last_processed_torrent_id'
+
+
+class TagRulesProcessor(TaskManager):
+    # this value must be incremented in the case of new rules set has been applied
+    version: int = 1
+
+    def __init__(self, notifier: Notifier, db: TagDatabase, mds: MDS.MetadataStore,
+                 batch_size: int = 1000, interval: float = 10):
```

Comment:
**Suggestion:** Extract these values to a global variable.

Reply:
Done

Reply:
Done.